### ISSUE TITLE
Batch trainer returns fake training loss values instead of real training

### ISSUE THREAD

**Issue by schancel** [darwin-finch/finch#11]
## Overview

`src/training/batch_trainer.rs` has a `// TODO: Actual training implementation` comment and returns placeholder loss values instead of actually training the model. This means the training pipeline appears to work but does nothing.

## Location

`src/training/batch_trainer.rs`, around line 228:
```rust
// TODO: Actual training implementation
// For now, return placeholder results
```

## Impact

- Users who trigger LoRA fine-tuning via Ctrl+G/Ctrl+B feedback see "training complete" messages
- No actual model adaptation occurs
- The feedback loop is broken — user feedback is collected but has no effect

## Context

The full LoRA training pipeline (issue #1, #7) requires significant work. This issue specifically tracks making the batch trainer:
1. Either properly implement training via the Python subprocess (call `scripts/train_lora.py`)
2. Or surface a clear error/warning so users know training isn't happening

## Recommended Short-term Fix

Wire up the existing Python training script:
```rust
// In batch_trainer.rs when batch is ready:
let result = LoRATrainingSubprocess::run(&examples, &config)?;
```

The Python script already exists at `scripts/train_lora.py` and handles weighted sampling.

## Files

- `src/training/batch_trainer.rs` — main fix location
- `scripts/train_lora.py` — Python training script (already exists)
- `src/training/subprocess.rs` — LoRATrainingSubprocess (may already exist)

## Acceptance Criteria

- [ ] Training actually runs when batch threshold is reached
- [ ] Real loss values are reported (or an honest "training skipped" message)
- [ ] No silent success on unimplemented code paths
